### ISSUE TITLE
Replacing Add Layer with Linear Combination of Learnable Weights

### ISSUE THREAD

**Issue by OUStudent** [tensorflow/tensorflow#55693]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Feature Request

The Add() Layer can be expressed as a linear combination of inputs where each coefficient value is 1. 

I was trying to make a custom layer that performs the same as Add, except uses trainable coefficients as we assume a coefficient of 1 is optimal. Unfortunately, I was unable to do so on my own, specifically the gradients. 

I do not think this would be too hard to implement as all the code is kind of already there, just need to calculate gradients and then update.

### Source

source

### Tensorflow Version

2.7

### Custom Code

Yes

### Current Behaviour?

```shell
The Add() Layer can be expressed as a linear combination of inputs where each coefficient value is 1. 

I was trying to make a custom layer that performs the same as Add, except uses trainable coefficients as we assume a coefficient of 1 is optimal. Unfortunately, I was unable to do so on my own, specifically the gradients. 

I do not think this would be too hard to implement as all the code is kind of already there, just need to calculate gradients and then update.
```

### Standlone code to reproduce the issue

```shell
x
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@OUStudent ,
Can you please elaborate about your Feature. Also, please specify the Use Cases for this feature. Thanks!

**Comment by OUStudent** [comment]
> @OUStudent , Can you please elaborate about your Feature. Also, please specify the Use Cases for this feature. Thanks!

Imagine we have a skip connection and a connection from the previous block. We can add these two together as follows:

skip = ....
prev = ....
next = Add()([skip, prev])

This add statement is mathematically equal to:
next = 1*skip + 1 * prev

As we can see, the coefficient for each input path is 1; however, perhaps we could train these coefficients to "learn" some type of attention mechanism :

next = alpha*skip + beta*next  

where alpha and beta are learned during training.

**Comment by sachinprasadhs** [comment]
Replacing an existing functionality will not be a good option since it breaks lots of user code. Rather we can consider it as a new feature request.
Since, development of keras moved to separate repository https://github.com/keras-team/keras/issues

Please post this issue on keras-team/keras repo.
To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!
